src/restClient.js

```diff
@@ -0,0 +1,719 @@
+     */
+    initialize() {
+        const restClient = this;
+        return new Promise((resolve, reject) => {
+            const that = this;
+            new Swagger({
+                url: `${config.sscAPIBase}/spec.json`,
+                usePromise: true,
+            }).then((api) => {
+                console.log("successfully loaded swagger spec. Attempting to call heartbeat /features ");
+                that.api = api;
+                async.waterfall([
+                    function getToken(callback) {
+                        restClient.generateToken("UnifiedLoginToken")
+                            .then((token) => {
+                                callback(null, token);
+                            }).catch((error) => {
+                                callback(error);
+                            });
+                    },
+                    function heartbeat(token, callback) {
+                        api["feature-controller"].listFeature({}, getClientAuthTokenObj(token)).then((features) => {
+                            callback(null, token);
+                        }).catch((error) => {
+                            callback(error);
+                        });
+                    }
+                ], function onDoneHandler(err, token) {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        that.token = token; //save token
+                        resolve("success");
+                    }
+                });
+
+            }).catch((err) => {
+                reject(err);
+            });
+
+        })
+    }
+    /**
+     * returns an array of {name: "controller name", apis: ["apifunc1", "apifunc2"...]}
+     */
+    getControllers() {
+        const restClient = this;
+        //get list of controllers
+        const controllersNames = Object.keys(restClient.api).filter((item) => {
+            if (restClient.api[item] === null) {
+                return false; //type of null is object
+            }
+            if (typeof restClient.api[item] === 'object' && restClient.api[item].apis) {
+                return true;
+            }
+            return false;
+        })
+        //attach to each controller list of funcitons.
+        const result = controllersNames.map((name) => {
+            let res = { name: name, apis: [] };
+            const attr = Object.keys(restClient.api[name]).filter((attr) => {
+                if (typeof restClient.api[name][attr] === 'function' && attr !== 'help') {
+                    return true;
+                }
+                return false;
+            });
+            res.apis = attr;
+            return res;
+        })
+        return result;
+    }
+    /**
+     * send for prediciton or training or future actions when implemented
+     * @param {*} versionId 
+     * @param {*} actionType 
+     */
+    sendToAA(versionId, actionType) {
```

Comment:
Since it is example for users who are not familiar with our abbreviations, it is better to use full names.

Reply:
updated comment to "send for Audit Assistant prediction or training or future actions when implemented"